src/main/java/fi/fmi/avi/converter/tac/swx/amd79/SWXAmd79TACParser.java

```diff
@@ -41,15 +41,20 @@ public class SWXAmd79TACParser extends AbstractTACParser<SpaceWeatherAdvisoryAmd
             LexemeIdentity.SWX_EFFECT_LABEL, LexemeIdentity.NEXT_ADVISORY, LexemeIdentity.REMARKS_START};
     private AviMessageLexer lexer;
 
-    private static Optional<PartialOrCompleteTimeInstant> createPartialTimeInstant(final Lexeme lexeme) {
+    private static Optional<PartialOrCompleteTimeInstant> createAnalysisTimeInstant(final Lexeme lexeme, @Nullable final PartialOrCompleteTimeInstant issueTime) {
         final Integer day = lexeme.getParsedValue(Lexeme.ParsedValueName.DAY1, Integer.class);
         final Integer minute = lexeme.getParsedValue(Lexeme.ParsedValueName.MINUTE1, Integer.class);
         final Integer hour = lexeme.getParsedValue(Lexeme.ParsedValueName.HOUR1, Integer.class);
 
-        if (day != null && minute != null && hour != null) {
-            return Optional.of(PartialOrCompleteTimeInstant.of(PartialDateTime.ofDayHourMinuteZone(day, hour, minute, ZoneId.of("Z"))));
+        if (day == null || minute == null || hour == null) {
+            return Optional.empty();
         }
-        return Optional.empty();
+
+        final PartialDateTime time = PartialDateTime.ofDayHourMinuteZone(day, hour, minute, ZoneId.of("Z"));
+        final PartialOrCompleteTimeInstant instant = issueTime != null && issueTime.getCompleteTime().isPresent()
+                ? PartialOrCompleteTimeInstant.of(time.toZonedDateTimeNear(issueTime.getCompleteTime().get()))
```

Comment:
Should we also set the partial time, so that it would be always present? Like

```suggestion
                ? PartialOrCompleteTimeInstant.of(time, time.toZonedDateTimeNear(issueTime.getCompleteTime().get()))
```